tests/components/google_assistant/test_trait.py

```diff
@@ -1117,6 +1117,10 @@ async def test_temperature_setting_climate_onoff(hass: HomeAssistant) -> None:
     )
     assert len(calls) == 1
 
+    # Reset config temperature_unit back to CELSIUS, required for
+    # additional tests outside this component.
+    hass.config.units.temperature_unit = UnitOfTemperature.CELSIUS
```

Comment:
I think we should reset it in the context manager that creates the `hass` instance instead. Here:
https://github.com/home-assistant/core/blob/688d5bb4c98130358e4cf9ae5ec8d30139cf6f19/tests/common.py#L346-L347

Reply:
Recreated https://github.com/home-assistant/core/pull/140295 to revert the changes here